src/applications/model/gaming-streaming-server.h

```diff
@@ -77,29 +94,31 @@ class GamingStreamingServer : public Application
 protected:
   /**
    * \brief Generate and send packets based on the different distribution
+   *
    * \param pktDist random variable which defines the size of generated packets
    * \param iatDist random variable which defines inter-arrival times between the packets
    */
   void SendPacket (Ptr<RandomVariableStream> pktDist, Ptr<RandomVariableStream> iatDist);
 
+  uint32_t     m_sentSeq;  //!< Sequence number for packets
+  uint32_t     m_totSentPackets; //!< Counter for sent packets
+  Ptr<Socket>  m_socket; //!< Socket
+  EventId      m_sendEvent; //!< Event to send the next packet
+  Address      m_peerAddress; //!< Remote peer address
+  uint16_t     m_peerPort; //!< Remote peer port
+
 private:
   virtual void StartApplication (void);
 
   virtual void StopApplication (void);
+
   /**
    * \brief Call the SendPacket to initiate an stream based on different random variables
    */
   virtual void Send (void);
 
-  uint32_t m_sentSeq;  //!< Sequence number for packets
-  uint32_t m_totSentPackets; //!< Counter for sent packets
-  Ptr<Socket> m_socket; //!< Socket
-  Address m_peerAddress; //!< Remote peer address
-  uint16_t m_peerPort; //!< Remote peer port
-  EventId m_sendEvent; //!< Event to send the next packet
-
-  Ptr<RandomVariableStream>  m_pks;  //!< Random number generator for packet size
-  Ptr<RandomVariableStream>  m_iat; //!< Random number generator for packets inter-arrival time
+  Ptr<RandomVariableStream>  m_packetSize;  //!< Random number generator for packet size
+  Ptr<RandomVariableStream>  m_interArrivalTime; //!< Random number generator for packets inter-arrival time
```

Comment:
as said before, you could generalize this class (and make it pure virtual) by defining directly `protected` `std::vector<RandomVariableStream>` directly here